directdemod/merger/merger.py

```diff
@@ -0,0 +1,256 @@
+            degree           = self.compute_angle(*bot, *top)
+
+            image_objects.append((image, center, degree))
+
+        return self.merge_images(image_objects, whole)
+
+    def update_extents(self, extent):
+
+        '''Update bounds of the projection
+
+        Args:
+            extent (:obj:`tuple(float, float, float, float)`): current image bounds
+        '''
+
+        self.left_ex  = min(self.left_ex,  extent[0])
+        self.right_ex = max(self.right_ex, extent[1])
+        self.bot_ex   = min(self.bot_ex,   extent[2])
+        self.top_ex   = max(self.top_ex,   extent[3])
+
+    def add_m(self, center, dx, dy):
+        # source: https://stackoverflow.com/questions/7477003/calculating-new-longitude-latitude-from-old-n-meters
+        new_latitude  = center[0] + (dy / 6371000.0) * (180 / np.pi)
+        new_longitude = center[1] + (dx / 6371000.0) * (180 / np.pi) / np.cos(center[0] * np.pi/180)
+        return (new_latitude, new_longitude)
+
+    @staticmethod
+    def extract_date(filename):
+
+        '''Extracts date from filename
+
+        Args:
+            filename (:obj:`string`): name of the file
+
+        Returns:
+            :obj:`datetime`: contructed datetime object
+        '''
+
+        parts = filename.split('_')
+        image_date, image_time = None, None
+        for index, part in reversed(list(enumerate(parts))):
+            if part[-1] == "Z":
+                image_time = part[:-1]
+                image_date = parts[index - 1]
+
+        if image_date is None or image_time is None:
+            raise ValueError("Invalid file name format.")
+
+        return ImageMerger.to_datetime(image_time, image_date)
+
+    @staticmethod
+    def to_datetime(image_time, image_date):
+
+        '''Builds datetime object
+
+        Args:
+            image_time (:obj:`string`): time when the image was captured
+            image_date (:obj:`string`): date when the image was captured
+
+        Returns:
+            :obj:`datetime`: contructed datetime object
+        '''
+
+        try:
+            year   = int(image_date[0:4])
+            month  = int(image_date[4:6])
+            day    = int(image_date[6:8])
+            hour   = int(image_time[0:2])
+            minute = int(image_time[2:4])
+            second = int(image_time[4:6])
+
+            return datetime(year, month, day, hour, minute, second)
+        except ValueError as e:
+            # add error logging
+            raise
+
+    def extract_coords(self, image, satellite, dtime):
+
+        '''Extracts coordinates of the image bounds
+
+        Args
+            image (:obj:`np.array`): captured image
+            satellite (:obj:`string`): name of the satellite
+            dtime (:obj:`datetime`): time when the image was captured
+
+        Returns:
+            :obj:`tuple`: extracted coordinates
+        '''
+
+        orbiter = Orbital(satellite) if self.tle_file is None else Orbital(satellite, tle_file=self.tle_file)
```

Comment:
Just to make sure: Please make sure that merger doesn't depend on this. Because, if it does it will become only NOAA specific. When we have Meteor images etc, it would be good if we can use those readily too. All this is already being done in NOAA part, you can save the results as I explained before and use it.

Reply:
Actully if we implement unified data file, where the image coordinated and other data would be store there won't be any need to compute it here. Everything will decoupled. And merger will only depend on the contents of the image descriptor file.

Reply:
Now, I have provided proper image storing and added specific method for NOAA images. This seems to be solved.